changelog/3299.feature.rst

```diff
@@ -0,0 +1 @@
+The ``--pdb`` option now causes KeyboardInterrupt to enter the debugger, instead of stopping the test session.
```

Comment:
Just to make sure: hitting CTRL+C again after landing on the debugger cases the debugger to exit right?

Reply:
Yes. I'll try to add a test that verifies that. Should I mention it in the changelog or docs?

Reply:
I think just mentioning it in the changelog is sufficient, something like:

> ... instead of stopping the test session. Hitting CTRL+C again exits the debugger.

Reply:
I see the test failed... not sure if it is easy to fix it; if not I'm OK with dropping the test, I think it is enough to ensure that we enter pdb when we get a `KeyboardInterrupt`.

Reply:
I also expected the test to fail because of pexpect's `kill` method not being cross-platform or something like that, but actually the test failed because pdb indeed behaves differently in python 3.6, and the KeyboardInterrupt does not cause it to exit. Are we ok with that?

Reply:
looks like it was an intentional change: https://bugs.python.org/issue7245.
Haha now I understand the discussion on the issue.

Reply:
The `nosigint` parameter to the Pdb constructor is supposed to disable that functionality, according to the docs (https://docs.python.org/3/library/pdb.html#pdb.Pdb). For some reason it doesn't seem to be working for me, but I can continue looking into it if we care about it.

On the other hand, probably better to let the pdb act like it usually acts for each version of python.

Reply:
> On the other hand, probably better to let the pdb act like it usually acts for each version of python.

Definitely, I agree! Thanks for researching this. 👍